---
authors: Michelle Gwinn, TIGR curators
id: "GO_REF:0000012"
year: 2003
layout: goref
---

## Pairwise alignment (TIGR)

Pairwise alignments are generated by taking two sequences and aligning them so that the maximum number of amino acids in each protein match, or are similar to, each other. Tools such as BLAST work by comparing a protein-of-interest individually with every protein in a database of known protein sequences and retaining only those matches with a high probability of being significant. Basic BLAST generates local alignments between proteins for regions of high similarity. Other pairwise alignment tools attempt to generate global (full-length) protein alignments. A tool called Blast_Extend_repraze (BER, http://ber.sourceforge.net) has some benefits over basic BLAST. Input into the BER tool includes the underlying DNA sequence for each protein as well as 300 nucleotides upstream and downstream of the predicted boundaries of the protein coding sequence. This allows annotators to see the DNA sequence that underlies the query protein as part of the alignment. In addition, the BER tool is able to look for continuation of regions of similarity through frameshifts and in-frame stop codons.  If such regions are found the alignment is continued. BER searches are done in a two-step process: step one is a BLAST search against a non-redundant protein database, significant BLAST hits are stored in a mini-database for each query protein; step two is a modified Smith-Waterman alignment between the query and the proteins in its mini-database. In order to assess whether a given BER alignment is good enough to assert that the query shares the function of the match protein, one must look at a several factors. First of all, the match protein must itself be experimentally characterized in order to avoid transitive annotation errors. In addition, any residues or secondary structures known to be important for function in the match protein must be conserved in the query. The alignment should be visually inspected to look for any areas of lesser quality that might indicate the two proteins do not share the same function. Although it is impossible to set cutoff values for percent identity and length of match that will apply for every alignment, there are some guidelines. In general at least 40% identity that extends over the full lengths of both proteins is required in order to even consider functional equivalence. However, this percentage is highly dependent on the length and complexity of the proteins. 40% identity between two proteins 500 amino acids long is much more significant that 40% identity between two proteins that are only 100 amino acids long. Therefore, the annotator's experience and knowledge of what is considered significant for the organism and protein family in question is very important. Some sets of proteins are much more highly conserved than others and therefore tolerances for percent identity may have to be adjusted. Finally, the alignment must be considered in the context of what else is known about the query protein and the organism as a whole.
